src/PeriodEnd/PaymentsDue/SFA.DAS.ProviderPayments.Calc.PaymentsDue/Domain/IDetermineWhichEarningsShouldBePaid.cs

```diff
@@ -0,0 +1,362 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using FastMember;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Dto;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Infrastructure.Data;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Infrastructure.Data.Entities;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Services.Dependencies;
+using SFA.DAS.ProviderPayments.Calc.PaymentsDue.Services.Extensions;
+
+namespace SFA.DAS.ProviderPayments.Calc.PaymentsDue.Domain
+{
+    // ReSharper disable once InconsistentNaming
+    public class IDetermineWhichEarningsShouldBePaid : IIDetermineWhichEarningsShouldBePaid
```

Comment:
is this an interface or a class? maybe the interface should be IDetermineWhichEarningsShouldBePaid.cs and the class something else.